edweiss-app/app/(app)/courses/[id]/archive.tsx

```diff
@@ -36,49 +45,52 @@ type AssignmentWithColor = Assignment & { color: Color; };
  * 
  * @returns JSX.Element - The rendered component for the previous assignments page.
  */
-const PreviousAssignmentsPage: ApplicationRoute = () => { //const PreviousAssignmentsPage: React.FC<PreviousAssignmentsProps> = ({ light, dark, extraInfo }) => {
-
-	const { extraInfo } = useLocalSearchParams();
+const ArchiveScreen: ApplicationRoute = () => {
+    const { extraInfo } = useLocalSearchParams();
 
-	let assignments: AssignmentWithColor[] = [];
+    let assignments: AssignmentWithColor[] = [];
 
+    if (typeof extraInfo === 'string') {
```

Comment:
Having this logic not wrapped inside a hook might be costly as it is executed at each re-render (`JSON.parse` is notoriously slow when used like that).
You can use the React hook [`useMemo`](https://react.dev/reference/react/useMemo) to optimize this.